directory/datadog_checks/directory/traverse.py

```diff
@@ -16,10 +16,7 @@ def _walk(top, follow_symlinks):
     dirs = []
     nondirs = []
 
-    try:
-        scandir_iter = scandir(top)
-    except OSError:
-        return
+    scandir_iter = scandir(top)
```

Comment:
Why do we need to remove this try-except? Doesn't seem related to the new logging statement.